logic/logic.go

```diff
@@ -79,3 +79,20 @@ func Load() ([]Dose, error) {
 		}
 	}
 }
+
+func Overwrite(doses []Dose) error {
+	file, err := os.OpenFile(dosageFilePath, os.O_WRONLY|os.O_TRUNC, 0600)
+	defer file.Close()
+
+	if err == nil {
+		for i := range doses {
+			if _, err = file.WriteString(doses[i].encode() + "\n"); err != nil {
+				break
+			}
+		}
+	}
+
+	file.Sync()
```

Comment:
Overwriting the existing file seems risky. A safer approach would be to write to a temporary file (e.g. in `/tmp`) and then move that.
```suggestion
	file, err := os.CreateTemp("", "skrive-tmp")
	if err == nil {
		err = file.Chmod(0600)
	}

	if err == nil {
		for i := range doses {
			if _, err = file.WriteString(doses[i].encode() + "\n"); err != nil {
				break
			}
		}
	}

	if err == nil {
		err = file.Sync()
	}
	
	closeErr := file.Close()
	if err == nil && closeErr != nil {
		return closeErr
	}

	if err == nil {
		err = os.Rename(file.Name(), dosageFilePath)
	}
```